packages/frontend/amp/server/document.tsx

```diff
@@ -53,10 +57,13 @@ export const document = ({
     <link rel="canonical" href="${metadata.canonicalURL}" />
     <meta name="viewport" content="width=device-width,minimum-scale=1">
     <link rel="icon" href="https://static.guim.co.uk/images/${favicon}">
-
-    <script type="application/ld+json">
-        ${JSON.stringify(linkedData)}
-    </script>
+    
+    ${linkedData.map(
+        ld => `
+<script type="application/ld+json" data-schema="${
+            ld['@type']
```

Comment:
How necessary is it to include the data-schema on the script tag? Reason I ask is because this line is potentially not typesafe if the object is missing a `@type` field.

Reply:
Ah seen you've introduced a new type (LinkedData) for this. You might find this complicates the data model extraction, but shout if I can help on that (or alternatively, if data-schema isn't necessary perhaps just get rid).

Reply:
ah, now that you've said that, it looks as if `data-schema` isn't required. I originally saw it used on theguardian.com. I've removed it now, along with the unnecessary type.